test/spanner.ts

```diff
@@ -2890,6 +2927,30 @@ describe('Spanner with mock server', () => {
       });
     });
 
+    it('should retry on aborted when running parallel query', async () => {
+      let attempts = 0;
+      const database = newTestDatabase();
+      const rowCount = await database.runTransactionAsync(
+        (transaction): Promise<number> => {
+          if (!attempts) {
+            spannerMock.abortTransaction(transaction);
+          }
+          attempts++;
+          return Promise.all([
+            transaction!.run(selectSql),
+            transaction!.run(selectSql),
+          ]).then(([rows]) => {
+            let count = 0;
+            rows.forEach(() => count++);
+            return transaction.commit().then(() => count);
+          });
+        }
+      );
+      assert.strictEqual(rowCount, 3);
+      assert.strictEqual(attempts, 2);
+      await database.close();
```

Comment:
Can we add verifications for the requests that we receive here? My expectation would be the following list of requests:

1. `ExecuteStreamingSql` (with `transaction.begin`)
2. `BeginTransaction`
3. `ExecuteStreamingSql` with transaction.id
4. `ExecuteStreamingSql` with transaction.id
5. `Commit`